akka-actor-typed-tests/src/test/java/akka/actor/typed/ManualTimerTest.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.actor.typed;
+
+//#manual-scheduling-simple
+import java.util.concurrent.TimeUnit;
+import static com.typesafe.config.ConfigFactory.parseString;
+
+import scala.concurrent.duration.Duration;
+
+import akka.actor.typed.javadsl.Actor;
+
+import org.junit.Test;
+
+import akka.testkit.typed.TestKit;
+import akka.testkit.typed.ExplicitlyTriggeredScheduler;
+import akka.testkit.typed.javadsl.TestProbe;
+
+public class ManualTimerTest extends TestKit {
+  ExplicitlyTriggeredScheduler scheduler;
+
+  public ManualTimerTest() {
+    super(parseString("akka.scheduler.implementation = \"akka.testkit.typed.ExplicitlyTriggeredScheduler\""));
+    this.scheduler = (ExplicitlyTriggeredScheduler) system().scheduler();
+  }
+
+  static final class Tick {}
+  static final class Tock {}
+
+  @Test
+  public void testScheduleNonRepeatedTicks() {
+    TestProbe<Tock> probe = new TestProbe<>(system(), testkitSettings());
+    Behavior<Tick> behavior = Actor.withTimers(timer -> {
+      timer.startSingleTimer("T", new Tick(), Duration.create(10, TimeUnit.MILLISECONDS));
+      return Actor.immutable( (ctx, tick) -> {
+        probe.ref().tell(new Tock());
+        return Actor.same();
+      });
+    });
+
+    spawn(behavior);
+
+    scheduler.expectNoMessageFor(Duration.create(9, TimeUnit.MILLISECONDS), probe);
```

Comment:
would it be more readable to put the `expectNoMessageFor` on the `probe`, with a `ExplicitlyTriggeredScheduler` parameter?

Reply:
I wanted to avoid confusion with `probe.expectNoMessage`, which will not automatically advance the time of the explicit scheduler. If we decide we want to have separate probes for normal vs manual schedulers anyway (see below) then it should indeed of course move to the probe.